src/resourcesSpec.ts

```diff
@@ -54,7 +56,11 @@ export class NoSuchResourceTypeAttribute extends CustomError {
 export function getResourceType(type: string){
     // If the type starts with Custom::, it's a custom resource.
     if(type.indexOf('Custom::') === 0){
-        return specification.ResourceTypes['AWS::CloudFormation::CustomResource']!;
+        // return the generic type if there's no such custom type defined
+        if (!specification.ResourceTypes[type]) {
+          type = 'AWS::CloudFormation::CustomResource';
+        }
+        return specification.ResourceTypes[type]!;
```

Comment:
what does the cheeky `!` do on the end of this?

Reply:
Nice catch, the cheeky `!` disables validation as *Typescript* believes that `type` can be `undefined` however, this should not occur in this particular scenario. I actually borrowed it from the previous code.